connector/google/google.go

```diff
@@ -257,17 +258,20 @@ func (c *googleConnector) getGroups(email string, fetchTransitiveGroupMembership
 		}
 
 		for _, group := range groupsList.Groups {
-			// TODO (joelspeed): Make desired group key configurable
-			userGroups = append(userGroups, group.Email)
-
-			// getGroups takes a user's email/alias as well as a group's email/alias
-			if fetchTransitiveGroupMembership {
-				transitiveGroups, err := c.getGroups(group.Email, fetchTransitiveGroupMembership)
-				if err != nil {
-					return nil, fmt.Errorf("could not list transitive groups: %v", err)
+			if _, exists := checkedGroups[group.Email]; !exists {
```

Comment:
NIT: I'd like to reduce the indentation level here, e.g., 

```go
if _, exists := checkedGroups[group.Email]; exists {
    continue
}
...
if !fetchTransitiveGroupMembership {
    continue
}
...    
```

Nested conditions wrapped in two for loops are a little bit harder to read.

Reply:
Perhaps use some tools in the build pipeline that do static analysis of cognitive and cyclomatic complexity.